spanner/spannertest/db_query.go

```diff
@@ -609,27 +571,68 @@ func (d *database) evalSelect(sel spansql.Select, qc *queryContext) (si *selIter
 					outRow = append(outRow, nil)
 				}
 			}
-			outRow = append(outRow, x)
+
+			for _, aggI := range aggI {
+				fexpr := sel.List[aggI].(spansql.Func)
+				fn := aggregateFuncs[fexpr.Name]
+				starArg := fexpr.Args[0] == spansql.Star
+				if starArg && !fn.AcceptStar {
+					return nil, fmt.Errorf("aggregate function %s does not accept * as an argument", fexpr.Name)
+				}
+				var argType spansql.Type
+				if !starArg {
+					ci, err := ec.colInfo(fexpr.Args[0])
+					if err != nil {
+						return nil, fmt.Errorf("evaluating aggregate function %s arg type: %v", fexpr.Name, err)
+					}
+					argType = ci.Type
+				}
+
+				// Compute aggregate value across this group.
+				var values []interface{}
+				for i := rg[0]; i < rg[1]; i++ {
+					ec.row = raw.rows[i]
+					if starArg {
+						// A non-NULL placeholder is sufficient for aggregation.
+						values = append(values, 1)
+					} else {
+						x, err := ec.evalExpr(fexpr.Args[0])
+						if err != nil {
+							return nil, err
+						}
+						values = append(values, x)
+					}
+				}
+				x, typ, err := fn.Eval(values, argType)
+				if err != nil {
+					return nil, err
+				}
+				aggType = typ
```

Comment:
This makes the type of all the aggregation expressions in the list equal to the type of the last expression. So the following statement will fail because it will try to interpret all the columns as INT64:

```sql
SELECT Cool, MIN(Name), MAX(Name), COUNT(*) FROM Staff GROUP BY Cool
```